tests/unittests/tests-ipv6_nc/tests-ipv6_nc.c

```diff
@@ -0,0 +1,397 @@
+{
+    ng_ipv6_addr_t addr = { {
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+        }
+    };
+
+    TEST_ASSERT_EQUAL_INT(-EINVAL, ng_ipv6_nc_add(KERNEL_PID_UNDEF, &addr,
+                          TEST_STRING4, sizeof(TEST_STRING4), 0));
+}
+
+static void test_ipv6_nc_add__EINVAL_ipv6_addr_unspecified(void)
+{
+    ng_ipv6_addr_t addr = NG_IPV6_ADDR_UNSPECIFIED;
+
+    TEST_ASSERT_EQUAL_INT(-EINVAL, ng_ipv6_nc_add(TEST_UINT16, &addr,
+                          TEST_STRING4, sizeof(TEST_STRING4), 0));
+}
+
+static void test_ipv6_nc_add__EINVAL_l2addr_too_long(void)
+{
+    ng_ipv6_addr_t addr = { {
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+        }
+    };
+
+    TEST_ASSERT_EQUAL_INT(-EINVAL, ng_ipv6_nc_add(TEST_UINT16, &addr,
+                          TEST_STRING4, NG_IPV6_NC_L2_ADDR_MAX + TEST_UINT8, 0));
+}
+
+static void test_ipv6_nc_add__ENOMEM(void)
+{
+    ng_ipv6_addr_t addr = { {
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+        }
+    };
+
+    for (int i = 0; i < NG_IPV6_NC_SIZE; i++) {
+        TEST_ASSERT_EQUAL_INT(0, ng_ipv6_nc_add(TEST_UINT16, &addr,
+                                                TEST_STRING4, sizeof(TEST_STRING4), 0));
+        addr.u16[7].u16++;
+    }
+
+    TEST_ASSERT_EQUAL_INT(-ENOMEM, ng_ipv6_nc_add(TEST_UINT16, &addr,
+                          TEST_STRING4, sizeof(TEST_STRING4), 0));
+}
+
+static void test_ipv6_nc_add__success(void)
+{
+    ng_ipv6_addr_t addr = { {
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+        }
+    };
+
+    TEST_ASSERT_EQUAL_INT(0, ng_ipv6_nc_add(TEST_UINT16, &addr,
+                                            TEST_STRING4, sizeof(TEST_STRING4), 0));
+
+    TEST_ASSERT_NOT_NULL(ng_ipv6_nc_get(TEST_UINT16, &addr));
+}
+
+static void test_ipv6_nc_remove__no_entry_pid(void)
+{
+    ng_ipv6_addr_t addr = { {
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+        }
+    };
+
+    test_ipv6_nc_add__success();
+    ng_ipv6_nc_remove(TEST_UINT16 + 1, &addr);
+    TEST_ASSERT_NOT_NULL(ng_ipv6_nc_get(TEST_UINT16, &addr));
+    TEST_ASSERT_NULL(ng_ipv6_nc_get(TEST_UINT16 + 1, &addr));
+}
+
+static void test_ipv6_nc_remove__no_entry_addr1(void)
+{
+    ng_ipv6_addr_t addr = { {
+            0x01, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+        }
+    };
+    ng_ipv6_addr_t exp_addr = { {
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+        }
+    };
+
+    test_ipv6_nc_add__success();
+    ng_ipv6_nc_remove(TEST_UINT16, &addr);
```

Comment:
Ah yes… but that's the point of this test: removing an address, that is not there. See 9986aef
 for clarification